test/integration/registerable_test.rb

```diff
@@ -238,6 +238,38 @@ def user_sign_up
     assert User.to_adapter.find_all.empty?
   end
 
+  test 'a signed in user should be able to cancel their account with current password if password required' do
+    swap Devise, require_password_to_destroy: true  do
+      sign_in_as_user
+      get edit_user_registration_path
+
+      within(".destroy-password") do
```

Comment:
I already mentioned the changes needed in the form, but just to add a comment here that ideally, for integration tests we won't include much implementation details - like a CSS class, for example - because that would made the test more fragile - e.g. if we change the CSS class the test would break, even though the system is still functioning the same way.